model/Signal.py

```diff
@@ -39,19 +39,57 @@ def get_features_window(self, x, list_features="all"):
             features = np.concatenate((features, f(x).reshape(1, 8)), axis=1)
         return features
 
-    def remove_transition(self):
-        s = self.get_features(list_features=["mav"]).sum(axis=1)
-        # detection
-        algo = rpt.Binseg(model="l2").fit(s)
-        result = algo.predict(n_bkps=1)[0]
-        if result > len(s)//2:
-            result = 20
-        trans_idx = result*self.step
-        if self.signal.shape[0]>2*trans_idx:
-            self.signal = self.signal[trans_idx:, :]
-            self.n_samples = self.signal.shape[0]
+    def remove_transition(self, algorithm="Binseg", w=50, min_s=30):
+        
+        if algorithm == "Binseg":
+
+            # detection using Binseg
+            s = self.get_features(list_features=["mav"]).sum(axis=1)
+            algo = rpt.Binseg(model="l2").fit(s)
+            result = algo.predict(n_bkps=1)[0]
+            if result > len(s) // 2:
+                result = 20
+            trans_idx = result * self.step
+            if self.signal.shape[0] > 2 * trans_idx:
+                self.signal = self.signal[trans_idx:, :]
+                self.n_samples = self.signal.shape[0]
+
+        if algorithm == "Pelt":
+
+            # store original signal
+            original_signal = self.signal.copy()
```

Comment:
.copy() makes a shallow copy. Both references still point to the same object in memory. So if you modify one, the other one gets modified as well. Also, don't modify the original signal.